core/dbt/utils.py

```diff
@@ -613,3 +615,34 @@ def _connection_exception_retry(fn, max_attempts: int, attempt: int = 0):
             _connection_exception_retry(fn, max_attempts, attempt + 1)
         else:
             raise ConnectionException('External connection exception occurred: ' + str(exc))
+
+
+def args_to_dict(args):
```

Comment:
can we leave a comment why this function is required? I'm not entirely sure what it's for.

Reply:
It's there because we get errors with some of the fields when dict-ifying, like PosixPath and the 'cls' key. This code was used to create the 'args' part of the run_results, and this just reuses that code.

Reply:
I've added a comment to the function.